.buildkite/scripts/upgrade_testing/upgrade_testing.js

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+const omit = require('lodash/omit');
+const execa = require('execa');
+const reduce = require('lodash/reduce');
+const find = require('lodash/find');
+const pRetry = require('p-retry');
+const axios = require('axios');
+const getConfig = require('./upgrade_testing_config');
+
+const CLOUD_API_KEY = process.env.CLOUD_API_KEY_SECRET;
+
+const request = axios.create({
+  baseURL: 'https://staging.found.no/api/v1/',
+  timeout: 10000,
+  headers: { Authorization: `ApiKey ${CLOUD_API_KEY}` },
+});
+
+const createInstance = async () => {
+  console.log('CLOUD_API_KEY', CLOUD_API_KEY);
+
+  let response;
+  try {
+    response = await request.post('deployments?validate_only=false', getConfig('7.15.0'));
+  } catch (error) {
+    console.log('error', error);
+    throw error;
+    // throw new pRetry.AbortError(error);
+  }
+
+
+  if (response.status === 404) {
+    throw new pRetry.AbortError(response.statusText);
+  }
+
+  return response;
+};
+
+const isInstanceReady = async ({ deploymentId }) => {
+  console.log('isInstanceReady');
+
+  const clusterInfo = await request.get(`/deployments/${deploymentId}`);
+
+  // console.log('clusterInfo', clusterInfo.data);
+
+  if (!clusterInfo.data?.healthy) {
+    throw new Error('Instance not ready yet');
+  }
+
+  return clusterInfo.data;
+};
+
+const upgradeInstance = async ({ deploymentId }) => {
+  console.log('upgradeInstance');
+  const clusterInfo = await isInstanceReady({ deploymentId });
+
+
+  const newConfig = {
+    prune_orphans: false,
+    resources: reduce(
+      clusterInfo.resources,
+      (acc, resource, key) => {
+        if (!resource.length) {
+          acc[key] = resource;
+          return acc;
+        }
+
+        // return value
+        acc[key] = resource.map((resourceItem) => ({
+          ...omit(resourceItem, ['info']),
+          plan: {
+            ...resourceItem.info.plan_info.current.plan,
+            [key]: {
+              ...resourceItem.info.plan_info.current.plan[key],
+              version: '7.16.0',
+            },
+          },
+        }));
+        return acc;
+      },
+      {}
+    ),
+  };
+
+  const response = await request.put(`deployments/${deploymentId}?validate_only=false`, newConfig);
+
+  await new Promise((r) => setTimeout(r, 30000));
```

Comment:
Why this timoeut? If there is a retry just below this I think we could get rid of this `setTimeout`, thoughts?